tensorrt_llm/_mnnvl_utils.py

```
@@ -180,6 +180,121 @@ def new_mnnvl_memory_address(cls, mapping: Mapping, size: int):
+            )
+
+        def _recv_fd(sock):
+            cmsg_space = socket.CMSG_SPACE(struct.calcsize("i"))
+            msg, anc, _, _ = sock.recvmsg(4, cmsg_space)
+            peer_rank = struct.unpack("i", msg[:4])[0]
+            for lvl, tp, data in anc:
+                if lvl == socket.SOL_SOCKET and tp == socket.SCM_RIGHTS:
+                    return peer_rank, struct.unpack("i", data[:4])[0]
+            raise RuntimeError("No SCM_RIGHTS ancillary data received")
+
+        sock_path = f"/tmp/.mnnvl_fd_{os.getpid()}"
```

Ruff (0.15.11)

[error] Probable insecure usage of temporary file or directory: "/tmp/.mnnvl_fd_"

(S108)